src/msg/simple/Accepter.cc

```diff
@@ -284,6 +289,45 @@ void Accepter::stop()
   done = false;
 }
 
+int Accepter::is_linked()
+{
+  entity_addr_t iaddr = msgr->get_myaddr();
 
-
-
+  struct ifaddrs *ifList;
+  if(getifaddrs(&ifList) < 0){
+    assert(0 == "unable to fetch interfaces and addresses");
+  }
+  
+  string ifname;
+  for (struct ifaddrs *ifa = ifList; ifa != NULL; ifa = ifa->ifa_next) {
+    entity_addr_t ifaddr;
+    ifaddr.set_sockaddr(ifa->ifa_addr);
+    if (iaddr.is_same_host(ifaddr)) {
+      ifname = std::string(ifa->ifa_name);
+      break;
+    }
+  }
+  freeifaddrs(ifList);
+  
+  if (ifname != string("")) {
+    struct ifreq ifr;
+    struct ethtool_value edata;
+    edata.cmd = ETHTOOL_GLINK;
+    edata.data = 0;
+    memset(&ifr, 0, sizeof(ifr));
+    strncpy(ifr.ifr_name, ifname.c_str(), ifname.length()+1);
+    ifr.ifr_data = (char *)&edata;
+  
+    int tfd = socket(iaddr.addr.ss_family, SOCK_DGRAM, 0 );
+    if (tfd < 0)
+      return -errno;;
+    if(ioctl( tfd, SIOCETHTOOL, &ifr ) == -1){
+      close(tfd);
+      return -EOPNOTSUPP;
+    }
```

Comment:
I'm having trouble finding documentation for this. What test exactly does this run?